src/mds/MDSMap.cc

```diff
@@ -663,69 +634,47 @@ void MDSMap::encode(bufferlist& bl, uint64_t features) const
     encode(session_autoclose, bl);
     encode(max_file_size, bl);
     encode(max_mds, bl);
-    __u32 n = mds_info.size();
-    encode(n, bl);
-    for (map<mds_gid_t, mds_info_t>::const_iterator i = mds_info.begin();
-	i != mds_info.end(); ++i) {
-      encode(i->first, bl);
-      encode(i->second, bl, features);
-    }
+    encode(mds_info, bl, features);
     encode(data_pools, bl);
     encode(cas_pool, bl);
 
     __u16 ev = 5;
     encode(ev, bl);
-    encode(compat, bl);
-    encode(metadata_pool, bl);
+    {
+      CompatSet _compat;
+      encode(_compat, bl);
+    }
+    encode((int64_t)0, bl);
     encode(created, bl);
     encode(modified, bl);
-    encode(tableserver, bl);
+    encode((int32_t)0, bl);
     encode(in, bl);
     encode(inc, bl);
     encode(up, bl);
     encode(failed, bl);
     encode(stopped, bl);
-    encode(last_failure_osd_epoch, bl);
```

Comment:
shouldn't remove last_failure_osd_epoch. If it belongs to ev part, encode a 0 here.  besides, when ev is 5.  encode up to  last_failure_osd_epoch